vnext/Desktop/DevSupportManager.cpp

```diff
@@ -55,60 +55,50 @@ string DevSupportManager::GetJavaScriptFromServer(
     const string &platform) {
   auto bundleUrl =
       DevServerHelper::get_BundleUrl(debugHost, jsBundleName, platform /*platform*/, "true" /*dev*/, "false" /*hot*/);
-
-  Url url(bundleUrl);
-  auto const resolveResult = m_resolver.resolve(url.host, url.port);
-  tcp::socket socket{m_context};
-  boost::system::error_code ec;
-  connect(socket, resolveResult, ec);
-  if (ec) {
+  Uri uri(Common::Unicode::Utf8ToUtf16(bundleUrl));
+  Filters::HttpBaseProtocolFilter filter;
+  filter.CacheControl().ReadBehavior(Filters::HttpCacheReadBehavior::NoCache);
+  HttpClient client(filter);
+  client.DefaultRequestHeaders().Host(HostName(uri.Host().c_str()));
+
+  HttpRequestMessage request(HttpMethod::Get(), uri);
+  HttpResponseMessage response = client.SendRequestAsync(request).get();
+
+  IBuffer buffer0 = response.Content().ReadAsBufferAsync().get();
+  auto reader = DataReader::FromBuffer(buffer0);
+  reader.UnicodeEncoding(UnicodeEncoding::Utf8);
+  auto length = reader.UnconsumedBufferLength();
+  string result;
+  if (length > 0 || response.IsSuccessStatusCode()) {
+    vector<uint8_t> data(length);
```

Comment:
>data [](start = 20, length = 4)

We should just be able to create string object with specific length, and fill that.
In this way you should be able to get rid of vector<uint8_t> object altogether

Reply:
Please note the `else` clause.
If an error status is obtained, we still return a string which will have a custom JSON message, unrelated to the response body:
```
stream << R"({"error:")"s << static_cast<int>(response.StatusCode()) << " downloading " << bundleUrl << R"("})"s;
```

Reply:
I'm not sure I follow, in the else case you end up with a string too, regardless of whether it's related to the response body or not... why do we need vector<uint8_t> ?

Reply:
We probably don’t. I based this on the UWP variant.
I’m not yet familiar with better practices for loading HTTP body content as strings.